### PR TITLE
BUG 90785 HttpWebRequest.Credentials property is cleared on redirect

### PR DESCRIPTION
**Pull request by stevewhims**
# Adding info about clearing HttpWebRequest.Credentials on redirect

## Summary

When a redirect takes place, HttpWebRequest.Credentials is cleared on under certain circumstances for security reasons. This change describes those circumstances (tracked in BUG 90785).

### CODE REVIEW

File: xml/System.Net/HttpWebRequest.xml [hunk 1]
```diff
@@ -1642,7 +1642,7 @@
   
  Supported authentication schemes include Digest, Negotiate, Kerberos, NTLM, and Basic.  
   
-   
+ For security reasons, credentials that you wish to be included when automatically following redirects (when the <xref:System.Net.HttpWebRequest.AllowAutoRedirect%2A> property is set to <see langword="true"/>) need to be stored within a <xref:System.Net.CredentialCache> assigned to this property. This property will automatically be set to <see langword="null"/> upon redirection if it contains anything except a <xref:System.Net.CredentialCache> or <xref:System.Net.WebClient.UseDefaultCredentials%2A>. Having this property value be automatically set to <see langword="null"/> under those conditions prevents credentials from being sent to any unintended destination.
```
Comment (davidsh) [inline xml/System.Net/HttpWebRequest.xml]: >or <xref:System.Net.WebClient.UseDefaultCredentials%2A>

This text should be removed.

**Review by davidsh — CHANGES_REQUESTED** [review]

File: xml/System.Net/HttpWebRequest.xml (hunk 1 above)
Comment (stevewhims) [reply xml/System.Net/HttpWebRequest.xml]: Thanks! Done.

File: xml/System.Net/HttpWebRequest.xml [hunk 2]
```diff
@@ -1642,7 +1642,7 @@
   
  Supported authentication schemes include Digest, Negotiate, Kerberos, NTLM, and Basic.  
   
-   
+ For security reasons, credentials that you wish to be included when automatically following redirects (when the <xref:System.Net.HttpWebRequest.AllowAutoRedirect%2A> property is set to <see langword="true"/>) need to be stored within a <xref:System.Net.CredentialCache> assigned to this property. This property will automatically be set to <see langword="null"/> upon redirection if it contains anything except a <xref:System.Net.CredentialCache>. Having this property value be automatically set to <see langword="null"/> under those conditions prevents credentials from being sent to any unintended destination.
```
Comment (mairaw) [inline xml/System.Net/HttpWebRequest.xml]: \<see langword="true"/> -> `true` (this section is the Markdown part)
\<see langword="null"/> -> `null`

The first sentence is a bit hard to parse. Maybe something like this would work better:
`For security reasons, when automatically following redirects (that is, when the <xref:System.Net.HttpWebRequest.AllowAutoRedirect> property is `true`), store the credentials that you want to include in a <xref:System.Net.CredentialCache> object? and assign it to this property.`

**Review by mairaw — CHANGES_REQUESTED** [review]
LGTM. I have a few suggestions for you.

**Review by mairaw — APPROVED** [review]
LGTM. Can't vouch for the technical accuracy of this though. 😄

**Review by davidsh — APPROVED** [review]
LGTM
